### PR TITLE
Remove VArray from GPUFieldDescBuilder

### PR DESCRIPTION
**Pull request by prybicki**

### CODE REVIEW

File: src/GPUFieldDescBuilder.hpp [hunk 1]
```diff
@@ -21,33 +21,49 @@
...
 #include <graph/Interfaces.hpp>
 #include <graph/NodesCore.hpp>
+#include <memory/HostPinnedArray.hpp>
+#include <memory/DeviceSyncArray.hpp>
 
 // Builder for GPUFieldDesc. Separated struct to avoid polluting gpu-visible header (gpu/GPUFieldDesc.hpp).
 struct GPUFieldDescBuilder
 {
-	static VArrayProxy<GPUFieldDesc>::Ptr buildReadable(const std::vector<std::pair<rgl_field_t, const void*>>& fieldsData)
+	static HostPinnedArray<GPUFieldDesc>::Ptr getHostBuffer()
```
Comment (msz-rai) [inline src/GPUFieldDescBuilder.hpp:30]: Could be private

**Review by msz-rai — COMMENTED** [review]
Good job!
Please address the comments. I have my doubts that it will work well in "full async mode".

File: src/GPUFieldDescBuilder.hpp [hunk 2]
```diff
@@ -21,33 +21,49 @@
...
 	{
-		auto gpuFields = GPUFieldDescBuilder::initialize(GPUFieldDescBuilder::getFields(fieldsData));
-		GPUFieldDescBuilder::fillWithData(gpuFields, fieldsData);
-		return gpuFields;
+		static auto hostBuffer = HostPinnedArray<GPUFieldDesc>::create();
```
[the remark this answers is not in the document]
Comment (prybicki) [reply src/GPUFieldDescBuilder.hpp:32]: Uhh, you're right — very naive approach. I will fix it.

**Review by msz-rai — APPROVED** [review]
Approved!

**Pull request merged**